### ISSUE TITLE
Custom tsconfig path

### ISSUE THREAD

**Issue by Akany** [Microsoft/TypeScript-Sublime-Plugin#542]
Is it possible to set plugin path to tsconfig ?

**Comment by DanielRosenwasser** [comment]
Yes, check out the repo's README [here](https://github.com/Microsoft/TypeScript-Sublime-Plugin/tree/0ca910635ac5927dd70bab359f80650da67b332f#note-using-different-versions-of-typescript).

**Comment by DanielRosenwasser** [comment]
Whoops, misread the issue - what exactly do you mean?

**Comment by Akany** [comment]
Im using angular cli app. It adds tsconfig in src folder - not in root.
Plugin doesnt see tsconfig.

**Comment by mhegazy** [comment]
@alexeagle is there a configuration to that can help move the tsconfig.json higher?

**Comment by alexeagle** [comment]
No, I believe the layout is fixed. But why is the plugin only looking in the root dir? Many projects keep their tsconfig.json under a `src` directory. I think this is a good idea, then you have a separate `test/tsconfig.json` since settings and deps are often different.
cc @hansl

**Comment by hansl** [comment]
Agreed with Alex. There are many reasons why having multiple tsconfig spread through a project is a good idea.

**Comment by jedmao** [comment]
In my case, I desire a separate tsconfig.json file for my `npm prepublish` command, which would allow me to transpile source files only (not test files) into a `./dist` dir with definitions, but not sourcemaps.

**Comment by FourwingsY** [comment]
Is this issue still not yet solved?

**Comment by KubaJastrz** [comment]
https://github.com/Microsoft/TypeScript-Sublime-Plugin/blob/26bb7c57a31fc1b62bfc50e8ec652f7ef3f49ff1/typescript/commands/build.py#L22 passes project directory as a `-p` argument to `tsc` command, so it looks only for default `tsconfig.json` filename. Would be very helpful to allow different file, or even have different configs per folder in project.

**Comment by rgant** [comment]
This is an issue for me in my Angular project because while I want to allow es6 in my test spec files I don't want that in my actual app. Angular supports this natively by having one tsconfig file for spec and another for the app. Would be nice if I could configure this plugin to support that.

**Comment by maciej-zuk** [comment]
Any updates on this one?

**Comment by dandee** [comment]
I've got the same issue.

**Comment by kraktus** [comment]
bump, same with Sublime 4

**Comment by jpike88** [comment]
Wow I wish I saw this issue before messing around. This makes TS development untenable aside from the most basic of scenarios.
